backend/deploy.js

```diff
@@ -34,4 +34,41 @@ exports.deployPages = function (socket, data) {
       socket.emit('deploy-success', {pagesURL: "https://" + data.username + ".github.io/" + repoName});
     }
   });
-}
+};
+
+exports.deployHeroku = function (socket, data) {
+  var spawn = require('child_process').spawn;
+
+  var email = data.email;
+  var herokuAPIKey = data.herokuAPIKey;
```

Comment:
would https://github.com/mick/passport-heroku and http://www.oyecode.com/2016/10/login-with-heroku-in-nodejs-applications.html help?